### PR TITLE
Add preliminary documentation for mousemap files.

### PR DESCRIPTION
**Pull request by Ultra-Instinct-05**
This PR adds some preliminary documentation for mouse map files.

Since mouse map files currently don't have the ability to define contexts, a warning quote has
also been added so that readers are aware of the current short comings.

### CODE REVIEW

File: docs/reference/mouse_bindings.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  This defines the number of times the button has to be pressed
+  for the corresponding `command` to trigger.
+
+
+::: warning
+Mousemap files don't have the ability to define contexts like key bindings,
+hence it is not advisable to ship them with packages or plugins, since
+the lack of context means the shipped mousemap files, taking over the 
+default mouse actions leading to undesirable behaviour.
+([#105](https://github.com/sublimehq/sublime_text/issues/105))
```
Comment (FichteFoll) [inline docs/reference/mouse_bindings.md:93]: The `:::` box is never closed.

Please also add some prose that brings the referenced issue into context, such as "This issue is being tracked at …".

File: docs/reference/mouse_bindings.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  mouse button, while the former is executed when the user releases the 
+  said button after it is pressed.
+
+`press_args`
+: The arguments for `press_command`.
+
+  This is a dictionary of arguments to be passed
+  on to the `press_command`.
+
+`count`
```
Comment (FichteFoll) [inline docs/reference/mouse_bindings.md:81]: The list of keys seems to be unsorted. I'd expect it to be sorted either conceptually (triggers first, then results) or alphabetically.

File: docs/reference/mouse_bindings.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  defines the right mouse button etc.
+  Similarly, if a command has to be executed when the scroll
+  wheel scrolls, then we have `scroll_up` & `scroll_down`.
+
+`modifiers`
+: A list of modifier keys.
+
+  This defines a list of modifiers keys that have to 
+  be pressed simultaneously (along with the corresponding button click)
+  for the command to be triggered. E.g. `["ctrl", "alt"]`.
```
Comment (FichteFoll) [inline docs/reference/mouse_bindings.md:52]: A list of supported modifiers or a reference to the key bindings page, which should have that already, would be appreciated.

File: docs/reference/mouse_bindings.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+in a mousemap file can take.
+
+`button`
+: The name of the button.
+
+  This defines the name of the button.
+  E.g. `button1` defines the left mouse button, `button2`
+  defines the right mouse button etc.
+  Similarly, if a command has to be executed when the scroll
+  wheel scrolls, then we have `scroll_up` & `scroll_down`.
```
Comment (FichteFoll) [inline docs/reference/mouse_bindings.md:45]: Please reword this to be more passive. Just explain that there are `button1` to `button16` (if memory serves me right) and `scroll_up` and `scroll_down` for the mouse-wheel.

We are in the reference section, which should be as precise and concise as possible.

File: docs/reference/mouse_bindings.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+---
+title: Mouse bindings
+---
+
+The mousemap files (which have the extension `.sublime-mousemap`)
+control what commands are executed when a user performs an action
+with a mouse e.g. clicking a mouse button, scrolling the scroll
+wheel etc.
+
```
Comment (FichteFoll) [inline docs/reference/mouse_bindings.md:9]: The semantic linefeeds in this paragraph are a bit off.

Please also mention that mousemaps use JSON.

**Review by FichteFoll — CHANGES_REQUESTED** [review]
Thanks for the additions. I left a few comments around some rough edges.

File: docs/reference/mouse_bindings.md (hunk 5 above)
Comment (Ultra-Instinct-05) [reply docs/reference/mouse_bindings.md:9]: TBH, I have not really understood the importance of semantic line feeds even after reading the linked URL in the CONTRIBUTING.md. If you can tell how the paragraph needs to be broken down or reworded, I'll do it.

File: docs/reference/mouse_bindings.md (hunk 2 above)
Comment (Ultra-Instinct-05) [reply docs/reference/mouse_bindings.md:83]: Can you explain a bit more ? Not really understood what do you mean ?

**Comment by Ultra-Instinct-05** [comment]
The other reviews should be resolved. Do let me know if there are any further issues.

File: docs/reference/mouse_bindings.md (hunk 2 above)
Comment (FichteFoll) [reply docs/reference/mouse_bindings.md:83]: I mean that there doesn't seem to be neither a group nor a consistent ordering for the keys explained here.

File: docs/reference/mouse_bindings.md (hunk 5 above)
Comment (FichteFoll) [reply docs/reference/mouse_bindings.md:9]: I'll do a manual pass on this later that you can review.

File: docs/reference/mouse_bindings.md (hunk 2 above)
Comment (Ultra-Instinct-05) [reply docs/reference/mouse_bindings.md:83]: As far as I know, there is no particular order in which the keys should appear. At least, in key bindings, you can reorder them in any way you want and they'd still work.

**Comment by FichteFoll** [comment]
Revised in d047e3bde955586ae2a033139671a897cb894d2f.
